module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <forward_list>
+#include <iostream>
+
+compressedImage compressGrayscale(const image& bitmap) {
+    compressedImage compression;
+
+    for (auto& it : bitmap) {
+        compression.emplace_back(std::make_pair(it.front(), 0));
+        for (auto it2 : it) {
+            std::pair<uint8_t, uint8_t>& pair = compression.back();
+            if (it2 == pair.first) {
+                pair.second++;
+            } else {
+                compression.emplace_back(std::make_pair(it2, 1));
```

Comment:
Did you know that you can make pair without calling std::make_pair? Like this:
compression.emplace_back(it2, 1);

Reply:
no, i didnt. i will try it some time.

Reply:
I didn't know too, but somebody gave me link to this thread in Stack Overflow in my program's code review:
https://stackoverflow.com/questions/53900865/emplace-back-vs-push-back-when-inserting-a-pair-into-stdvector